R/tribble.R

```diff
@@ -27,19 +27,75 @@
 #'   "b", 4:6
 #' )
 tribble <- function(...) {
+  data <- extract_frame_data_from_dots(...)
 
+  if (length(data$frame_rest) == 0) {
+    out <- rep(list(logical()), length(data$frame_names))
+    names(out) <- data$frame_names
+    return(as_tibble(out))
+  }
+
+  frame_mat <- matrix(data$frame_rest, ncol = length(data$frame_names), byrow = TRUE)
+
+  frame_col <- lapply(seq_len(ncol(frame_mat)), function(i) {
```

Comment:
Could you please extract everything after `extract_frame_data_from_dots(...)` to a separate function, and perhaps also `function(i) {...}` ?